src/openzeppelin/token/erc20/library.cairo

```diff
@@ -34,291 +29,335 @@ end
 end
 
 @storage_var
-func ERC20_allowances(owner: felt, spender: felt) -> (allowance: Uint256):
-end
-
-#
-# Constructor
-#
-
-func ERC20_initializer{
-        syscall_ptr : felt*, 
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }(
-        name: felt,
-        symbol: felt,
-        decimals: felt
-    ):
-    ERC20_name_.write(name)
-    ERC20_symbol_.write(symbol)
-    with_attr error_message("ERC20: decimals exceed 2^8"):
-        assert_lt(decimals, UINT8_MAX)
-    end
-    ERC20_decimals_.write(decimals)
-    return ()
-end
-
-func ERC20_name{
-        syscall_ptr : felt*,
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }() -> (name: felt):
-    let (name) = ERC20_name_.read()
-    return (name)
-end
-
-func ERC20_symbol{
-        syscall_ptr : felt*,
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }() -> (symbol: felt):
-    let (symbol) = ERC20_symbol_.read()
-    return (symbol)
-end
-
-func ERC20_totalSupply{
-        syscall_ptr : felt*, 
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }() -> (totalSupply: Uint256):
-    let (totalSupply: Uint256) = ERC20_total_supply.read()
-    return (totalSupply)
-end
-
-func ERC20_decimals{
-        syscall_ptr : felt*, 
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }() -> (decimals: felt):
-    let (decimals) = ERC20_decimals_.read()
-    return (decimals)
-end
-
-func ERC20_balanceOf{
-        syscall_ptr : felt*, 
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }(account: felt) -> (balance: Uint256):
-    let (balance: Uint256) = ERC20_balances.read(account)
-    return (balance)
-end
-
-func ERC20_allowance{
-        syscall_ptr : felt*, 
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }(owner: felt, spender: felt) -> (remaining: Uint256):
-    let (remaining: Uint256) = ERC20_allowances.read(owner, spender)
-    return (remaining)
+func allowances_(owner: felt, spender: felt) -> (allowance: Uint256):
 end
 
-func ERC20_transfer{
-        syscall_ptr : felt*, 
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }(recipient: felt, amount: Uint256):
-    let (sender) = get_caller_address()
-    _transfer(sender, recipient, amount)
-    return ()
-end
-
-func ERC20_transferFrom{
-        syscall_ptr : felt*, 
-        pedersen_ptr : HashBuiltin*,
-        range_check_ptr
-    }(
-        sender: felt, 
-        recipient: felt, 
-        amount: Uint256
-    ) -> ():
-    alloc_locals
-    let (caller) = get_caller_address()
-    let (caller_allowance: Uint256) = ERC20_allowances.read(owner=sender, spender=caller)
-
-    with_attr error_message("ERC20: transfer amount exceeds allowance"):
-        let (new_allowance: Uint256) = uint256_checked_sub_le(caller_allowance, amount)
+namespace ERC20:
+
+    #
+    # Constructor
+    #
+
+    func constructor{
+            syscall_ptr : felt*,
+            pedersen_ptr : HashBuiltin*,
```

Comment:
No strong opinions, although not sure if it counts as a constructor since it's not initializing a contract but a subset. Maybe `initializer` helps people distinguish between these two but again, no strong opinions.